File: api/api_studytutorialonline/app/Models/SinhVien.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use App\Models\Lop;
use App\Models\CTBaiTap;
use App\Models\DS_SinhVien;
use App\Models\TraLoi;
use Laravel\Sanctum\HasApiTokens;

class SinhVien extends Model
{
    use HasFactory;
    use HasApiTokens;
    protected $table = 'sinh_viens';
    protected $fillable = [
        'id',
        'id_khoa',
        'id_chuc_vu',
        'id_lop',
        'email',
        'password',
        'avt',
        'ma_so',
        'sdt',
        'ho_ten',
        'ngay_sinh',
        'trang_thai'
    ];
    public function lop()
    {
        return $this->beLongsTo(Lop::class, 'id_lop');
    }
    public function ctbaitap()
    {
        return $this->hasMany(CTBaiTap::class, 'id_sinh_vien', 'id');
    }
    public function dssinhvien()
    {
        return $this->hasMany(DS_SinhVien::class, 'id_sinh_vien', 'id');
    }
    public function traloi()
    {
        return $this->hasMany(TraLoi::class, 'id_sinh_vien', 'id');
    }
    public function binhluan()
    {
        return $this->hasMany(BinhLuan::class, 'id_sinh_vien', 'id');
    }
    public function baiviet()
    {
        return $this->hasMany(BaiViet::class, 'id_sinh_vien', 'id');
    }
}

File: app/user_flutter/lib/data/model/sinhVien.dart
class SinhVien {
  final int id;
  final String? id_lop;
  final String? email;
  final String? password;
  final String? ho_ten;
  final String? avt;
  final String? mssv;
  final String? sdt;
  final String? ngay_sinh;
  final int trangthai;
  final String token;

  const SinhVien(
      {required this.id,
      required this.id_lop,
      required this.email,
      required this.password,
      required this.ho_ten,
      required this.avt,
      required this.mssv,
      required this.ngay_sinh,
      required this.sdt,
      required this.trangthai,
      required this.token});
  factory SinhVien.fromJson(Map<String, dynamic> json) {
    return SinhVien(
        id: json['sinhvien']['id'],
        id_lop: json['sinhvien']['id_lop'],
        email: json['sinhvien']['email'],
        password: json['sinhvien']['password'],
        ho_ten: json['sinhvien']['ho_ten'],
        avt: json['sinhvien']['avt'],
        mssv: json['sinhvien']['mssv'],
        ngay_sinh: json['sinhvien']['ngay_sinh'],
        sdt: json['sinhvien']['sdt'],
        trangthai: json['sinhvien']['trang_thai'],
        token: json['token']);
  }
}
